client/postit/src/App.js

```diff
@@ -4,8 +4,11 @@ import SignUp from "./components/SignUp";
 import SignIn from "./components/SignIn";
 import API_ROUTES from "./utils/constants";
 import { useState } from "react";
+import { usePromiseTracker } from "react-promise-tracker";
```

Comment:
Why are you using this promise tracker?, I would eliminate the extra dependency, you can just use normal javascript and react just use a variable in state, and update it accordingly, i.e when data is returned  and so on